paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -17,33 +22,87 @@ import javax.inject.Inject
 @CustomerSessionScope
 @Suppress("unused")
 internal class CustomerSheetViewModel @Inject constructor(
+    application: Application,
     private val configuration: CustomerSheet.Configuration,
     private val customerAdapter: CustomerAdapter,
     private val lpmRepository: LpmRepository,
-) : ViewModel() {
+) : AndroidViewModel(application) {
 
     private val _viewState = MutableStateFlow<CustomerSheetViewState>(CustomerSheetViewState.Loading)
     val viewState: StateFlow<CustomerSheetViewState> = _viewState
 
+    private val _action = MutableStateFlow<CustomerSheetAction?>(null)
+    val action: StateFlow<CustomerSheetAction?> = _action
+
     init {
+        loadPaymentMethods()
+    }
+
+    private fun loadPaymentMethods() {
         viewModelScope.launch {
-            delay(DELAY)
-            _viewState.update {
-                CustomerSheetViewState.SelectPaymentMethod(
-                    configuration.headerTextForSelectionScreen
-                )
+            customerAdapter.retrievePaymentMethods().mapCatching {
+                val paymentOptionItems = it.map { paymentMethod ->
+                    PaymentOptionsItem.SavedPaymentMethod(
+                        displayName = paymentMethod
+                            .getLabel(getApplication<Application>().resources)
+                            .toString(),
+                        paymentMethod = paymentMethod
+                    )
+                }
+                customerAdapter.retrieveSelectedPaymentOption().mapCatching { selected ->
```

Comment:
Can we parallelize this instead of doing it sequentially (and does that even give us much)?